cmd/core/service/resource_manager.go

```diff
@@ -23,10 +23,33 @@ type IPAM struct {
 	sync.RWMutex
 }
 
-func NewResourceManager(ftup bool, teidRange uint32) (*ResourceManager, error) {
+func NewResourceManager(ueip, ftup bool, ipRange string, teidRange uint32) (*ResourceManager, error) {
 
+	var ipam IPAM
 	var fteidm FTEIDM
 
+	if ueip {
+		_, ipNet, err := net.ParseCIDR(ipRange)
+		if err != nil {
+			return nil, err
+		}
+
+		freeIPs := make([]net.IP, 0, 10000)
```

Comment:
Why 10000?

Reply:
with the current value of ip_pool (but it may be different), 65536 IP addresses will be generated. I thought it would be good to allocate 10k right away when declaring the slice to minimize the number of memory allocations. in our case we can immediately set the value to 65536: make([]net.IP, 0, 65536)